core/drivers/scmi-msg/clock_generic.c

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2021, Microchip
+ */
+
+#include <drivers/clk.h>
+#include <drivers/scmi.h>
+#include <drivers/scmi-msg.h>
+#include <kernel/boot.h>
+#include <string.h>
+#include <sys/queue.h>
+
+#define SCMI_MAX_CLK_NAME_LEN	16
+
+/**
+ * struct scmi_clk - Binds an SCMI channel/clock to a core clk reference
+ * @clk:        Core clock reference
+ * @channel_id: SCMI server channel handler exposing the clock
+ * @scmi_id:    SCMI clock domain ID
+ * @enabled:    SCMI clock state
+ * @link:       Reference in SCMI server clock list
+ */
+struct scmi_clk {
+	struct clk *clk;
+	unsigned int channel_id;
+	unsigned int scmi_id;
+	bool enabled;
+	SLIST_ENTRY(scmi_clk) link;
+};
+
+static bool scmi_clk_init_done;
+static SLIST_HEAD(, scmi_clk) scmi_clk_list =
+					SLIST_HEAD_INITIALIZER(scmi_clk_list);
```

Comment:
nitpicking: one or 2 tabs should be sufficient

Reply:
Done